### PR TITLE
Support wagtail42

### PR DESCRIPTION
**Pull request by nickmoreton**
This updates the package for Wagtail 4.2

## Additions

Wagtail 4.1

-

## Removals

Wagtail < 4.1

-

## Changes
-

## Testing

Updates github actions and tox matrix

-

## Review

- @user

[Preview this PR without the whitespace changes](?w=0)

## Notes

-

## Todos

-

## Checklist

* [x] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [x] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -2,6 +2,10 @@ name: test
 
 on: [pull_request, push]
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
```
Comment (nickmoreton) [inline .github/workflows/test.yml:8]: Cancels previous running actions if not finished.

File: wagtailsharing/models.py [hunk 2]
```diff
@@ -56,7 +50,7 @@ def root_url(self):
 class ShareableRoutablePageMixin(RoutablePageMixin):
     def route(self, request, path_components):
         if getattr(request, "routed_by_wagtail_sharing", False):
-            page = self.get_latest_revision_as_page()
+            page = self.get_latest_revision_as_object()
```
Comment (nickmoreton) [inline wagtailsharing/models.py:53]: Changed in Wagtail 4.0

File: wagtailsharing/views.py [hunk 3]
```diff
@@ -95,7 +87,7 @@ def serve(page, request, args, kwargs):
                 return result
 
         # Get the latest revision for the requested page.
-        page = page.get_latest_revision_as_page()
+        page = page.get_latest_revision_as_object()
```
Comment (nickmoreton) [inline wagtailsharing/views.py:90]: Changed in Wagtail 4.0

**Review by victoriachan — CHANGES_REQUESTED** [review]
Hi Nick, I have reviewed this. It looks good. The Readme needs updating though. I have attached some new screenshots to save you the trouble. 

I wasn't sure if the Page.serve() changes would affect this, so I tried this locally and it works fine.

File: README.rst [hunk 4]
```diff
@@ -202,9 +202,9 @@ Compatibility
 
 This project has been tested for compatibility with:
```
[the remark this answers is not in the document]
Comment (nickmoreton) [reply README.rst:211]: Thanks @victoriachan I have updated the docs.

And thanks so much for the screen shots too 👍

[the remark this answers is not in the document]
Comment (victoriachan) [reply README.rst:211]: Looks good!

**Comment by victoriachan** [comment]
Looks good to me!

**Review by victoriachan — APPROVED** [review]

**Comment by willbarton** [comment]
@nickmoreton It looks like your upgrade commits are relatively independent of the token commits. I'll pull them into a separate branch and get it ready to merge!

**Pull request closed without merging**